src/pages/user/Login.page.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Button } from "@components";
+import { useAuth } from "@providers";
+
+const UserLoginPage = () => {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [error, setError] = useState<null | unknown>(null);
+
+    const authProvider = useAuth();
+    const navigate = useNavigate();
+
+    const login = async (email: string, password: string) => {
+        try {
+            await authProvider.login(email, password);
+            navigate("/user", { replace: true });
+        } catch (error) {
+            setError(error);
+        }
+    };
+
+    const handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        login(email, password);
```

Comment:
There is room for refactoring here. You can remove this line and move everything in `login()`. It would do the same thing, you just need to make the handler async.